### PR TITLE
Geocoding source

### PR DESCRIPTION
**Pull request by cristineguadelupe**
Screenshot 2022-11-03 at 14 14 17
Screenshot 2022-11-03 at 14 12 02

https://user-images.githubusercontent.com/5660941/199791791-7d7718c3-e1e5-489c-8e70-d749c1b2544d.mov

### CODE REVIEW

File: lib/kino_maplibre/map_cell.ex [hunk 1]
```diff
@@ -11,6 +11,9 @@ defmodule KinoMapLibre.MapCell do
   @geometries [Geo.Point, Geo.LineString, Geo.Polygon, Geo.GeometryCollection]
   @styles %{"street (non-commercial)" => :street, "terrain (non-commercial)" => :terrain}
 
+  @query_source %{columns: nil, type: "query", variable: "Geocoding source"}
```
Comment (josevalim) [inline lib/kino_maplibre/map_cell.ex:14]: I think we don't need "source" on the name. I was also wondering if we could annotate this especially, so it is clear to users it is not a variable. WDYT about:

```suggestion
  @query_source %{columns: nil, type: "query", variable: "🌎 Geocoding"}
```

File: test/kino_maplibre/map_cell_test.exs [hunk 2]
```diff
@@ -51,7 +53,8 @@ defmodule KinoMapLibre.MapCellTest do
       %{columns: ["latitude", "longitude", "mag"], type: "table", variable: "earthquakes"},
       %{columns: ["coordinates", "year"], type: "table", variable: "conferences"},
       %{columns: nil, type: "geo", variable: "point"},
-      %{columns: nil, type: nil, variable: "quakes"}
+      %{columns: nil, type: nil, variable: "quakes"},
+      %{columns: nil, type: "query", variable: "Geocoding source"}
```
Comment (josevalim) [inline test/kino_maplibre/map_cell_test.exs:57]:
```suggestion
      %{columns: nil, type: "query", variable: "🌎 Geocoding"}
```

File: lib/assets/map_cell/main.js [hunk 3]
```diff
@@ -128,6 +128,25 @@ export function init(ctx, payload) {
                       />
                     </template>
                   </template>
+                  <template v-if="layer.source_type === 'query'" >
+                    <BaseInput
+                      name="layer_source_query"
+                      label="Geocode"
```
Comment (josevalim) [inline lib/assets/map_cell/main.js:134]: Do we support hints/placeholders? It would be nice to say something like: "Type a city, state, or country"

Comment (cristineguadelupe) [reply lib/assets/map_cell/main.js:134]: Yep!

**Review by josevalim — APPROVED** [review]

**Pull request merged**